docs/adr/0045-testing-updates-and-recommendations.md

```diff
@@ -0,0 +1,233 @@
+
+Additionally, while valuable, the functional test suite can consume significant CI resources and any effort to trim it back can help. A number of these tests could be more efficiently executed at the unit or integration test level. And, a [recent discovery][invalid test thread] found that at least one set of tests were running as false positives, not actually testing what we thought they were.
+
+Similarly, the [test result dashboard][test metrics dashboard] that provides insight into metrics like run times and skip rates is partially broken and has not been accurately reporting since late February or early March on a few metrics. This reduces visibility into test health and performance, and provides a false positive on health.
+
+Finally, documentation for testing _does_ exist, but some of it is outdated and does not include modern practices or guidance on designing tests. There is a lack of centralized best practices for when to use different types of tests (unit, integration, functional), identifying common code smells (e.g., excessive mocking, too many assertions), or how to standardize test structure and execution patterns across sub-projects. The knowledge that does exist lives fragmented in different locations, making onboarding and cross-team collaboration more difficult.
+
+All of these are things that we should probably tackle at some point, but the level of implementation, order, and priority are yet to be understood.
+
+## Decision Drivers
+
+- Reduce time and resource cost of test runs, particularly in CI
+- Improve developer experience and maintainability of tests
+- Increase reliability and clarity of test results
+- Promote consistency in framework usage, naming conventions, and structure
+- Establish central documentation and best practices
+- Improve ability to debug and diagnose test failures
+
+## Considered Options
+
+Given there are a few parallel ideas in this, we'll break them down individually, just expand each section to checkout more details!
+
+<details>
+  <summary>A: Unifying Testing Frameworks</summary>
+
+Options:
+
+1. Migrate All Projects to a Single Framework (Jest)
+2. Pilot Migration in Select Projects **👍 Recommended**
+3. Maintain Status Quo
+
+#### Details
+
+There is a divide in the test frameworks and how the run across all of the FxA monorepo. The vast majority of tests are split between using `Jest` and `Mocha`, with most new projects having been implemented with `Jest`. Migrating all tests (where possible) to use a single framework gains us some wins outlined below, however this would be a large undertaking, requiring several weeks of work including testing (ha), and cleanup of legacy setup to run `Mocha` or other runner based tests.
```

Comment:
```suggestion
There is a divide in the test frameworks and how they run across all of the FxA monorepo. The vast majority of tests are split between using `Jest` and `Mocha`, with most new projects having been implemented with `Jest`. Migrating all tests (where possible) to use a single framework gains us some wins outlined below, however this would be a large undertaking, requiring several weeks of work including testing (ha), and cleanup of legacy setup to run `Mocha` or other runner based tests.
```